server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java

```diff
@@ -161,40 +164,29 @@ private RrtExecutionResult getRrtExecutionResult(Mail mail, MailAddress recipien
             .collect(Guavate.toImmutableList());
     }
 
-    private ImmutableList<MailAddress> buildMailAddressFromMappingAddress(ImmutableList<Mapping> newMappings) {
-        return newMappings.stream()
-            .map(mailAddressFromMapping)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .collect(Guavate.toImmutableList());
-    }
+    private ImmutableList<Mapping> sanitizeMappingsWithNoDomain(Mappings mappings, DomainList domainList) throws MessagingException {
+        Supplier<Domain> defaultDomainSupplier = Suppliers.memoize(
```

Comment:
It would be cleaner to provide a memoize that doesn't require a guava supplier

Reply:
Do you have something in mind?

Reply:
yes, we could hide the Guava usage behind the method call

Reply:
What I did was implementing my own MemoizedSupplier. Feel free to update this.